### PR TITLE
feat: include resources with kubernetes.io prefix.

### LINKED ISSUE #1939: Do not collect  resources that start with ’kubernetes.io/‘
**What happened**:

my pod
```
limits:
        kubernetes.io/batch-cpu: 2k
        kubernetes.io/batch-memory: 4Gi
      requests:
         kubernetes.io/batch-cpu: "100"
         kubernetes.io/batch-memory: 819Mi
```
but not to be collected.

reason is  ’kubernetes.io/‘ Is considered to be a native resource:
https://github.com/kubernetes/kube-state-metrics/blob/main/internal/store/utils.go#L155

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- kube-state-metrics version: 
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Other info:

### PR DESCRIPTION
**Pull request by fengyehong**
Fixes #1939

### CODE REVIEW

File: internal/store/utils.go [hunk 1]
```diff
@@ -169,7 +169,7 @@ func isNativeResource(name v1.ResourceName) bool {
 }
 
 func isPrefixedNativeResource(name v1.ResourceName) bool {
-	return strings.Contains(string(name), v1.ResourceDefaultNamespacePrefix)
+	return false
 }
```
Comment (logicalhan) [inline internal/store/utils.go:173]: If we're just going to return false, then we should just get rid of this method.

Comment (logicalhan) [reply internal/store/utils.go:173]: Also there should be a test for this, we shouldn't just be able to change this function without some test breaking.

Comment (dgrisonnet) [reply internal/store/utils.go:173]: I think we should remove the code that check whether the resource is native or not as a whole.

The idea would be to make `Integer` the default unit if nothing is matched after https://github.com/kubernetes/kube-state-metrics/blob/d3afe69fd784b7ea31a42ce2eaceb7f7b1ead7a0/internal/store/pod.go#L193-L204 and remove the condition for extended resources: https://github.com/kubernetes/kube-state-metrics/blob/d3afe69fd784b7ea31a42ce2eaceb7f7b1ead7a0/internal/store/pod.go#L205-L211

**Comment by eahydra** [comment]
ping @fengyehong

File: internal/store/utils.go (hunk 1 above)
Comment (fengyehong) [reply internal/store/utils.go:169]: updated, i noticed there is already a test case for "kubernetes.io/resource-foo" so just updated expectVal

**Comment by eahydra** [comment]
@dgrisonnet @logicalhan PTAL

**Pull request closed without merging**